src/transforms/kubernetes/mod.rs

```diff
@@ -1 +1,585 @@
 pub mod watch_client;
+
+use self::watch_client::{ClientConfig, PodEvent, RuntimeError, Version, WatchClient};
+use super::Transform;
+use crate::{
+    event::{self, Event, Value},
+    sources::kubernetes::POD_UID,
+    topology::config::{DataType, TransformConfig, TransformContext, TransformDescription},
+};
+use bytes::Bytes;
+use evmap::{ReadHandle, WriteHandle};
+use futures::{
+    compat::{Future01CompatExt, Stream01CompatExt},
+    future::{select, Either},
+    stream::StreamExt,
+};
+use k8s_openapi::api::core::v1::Pod;
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+use snafu::{ResultExt, Snafu};
+use std::collections::{BTreeMap, VecDeque};
+use std::time::{Duration, Instant};
+use string_cache::DefaultAtom as Atom;
+use tokio::timer::Delay;
+
+// *********************** Defined by Vector **************************** //
+
+/// Node name `spec.nodeName` of Vector pod passed down with Downward API.
+const NODE_NAME_ENV: &str = "VECTOR_NODE_NAME";
+
+/// 1h sounds a lot, but a metadata entry averages around 300B and in an extreme
+/// scenario when a pod is deleted every second, we would have ~1MB of probably
+/// not usefull data. Which is acceptable. The benefit of such a long delay
+/// is that we will certainly* have the metadata while we are processing the
```

Comment:
What's the `*` on the `certainly*`?

Reply:
`certainly` in all normal situations. An abnormal would be:
 - Node is run in a VM which was paused for an hour.
 - Large disk i/o and large amount of logs that need to be picked up, and ram overflowed to swap.
 - etc.